inc/alert.class.php

```diff
@@ -231,7 +231,8 @@ public static function findAllToNotify($params = []) {
       }
 
       if ($p['show_hidden_alerts']) {
-         $login_show_hidden_sql = " `$utable`.`id` IS NOT NULL ";
+         //dont show hidden alert if they should no longer be visible
+         $login_show_hidden_sql = " `$utable`.`id` IS NOT NULL AND ( `$table`.`is_displayed_onhelpdesk`='1' OR `$table`.`is_displayed_onlogin`='1')";
```

Comment:
I do not understand why there are new conditions here.

Reply:
When an alert is hidden (as a result of being closed by the user) (see option can close the alert)
![image](https://user-images.githubusercontent.com/7335054/108675389-0845ed80-74e7-11eb-9d71-922cb1b8d0f0.png)


if this alert is no longer displayed, 
![image](https://user-images.githubusercontent.com/7335054/108675427-1136bf00-74e7-11eb-8ba8-b75a3231165b.png)

it is still displayed as a hidden alert
![image](https://user-images.githubusercontent.com/7335054/108675465-1f84db00-74e7-11eb-8e54-e6387fe44119.png)

Reply:
Unless I am wrong, all alerts should be displayed on central page, even when they ane not configured to be displayed on login page and helpdesk page.

Reply:
@orthagh Is this the desired behaviour on the central page ?

Reply:
> Unless I am wrong, all alerts should be displayed on central page, even when they ane not configured to be displayed on login page and helpdesk page.

Yes, unless they are disabled.

There is no switch (apart is_active) currently to choose to display alert everywhere but not on central page.

- They are always displayed on central page
- You can choose to display them also on login page
- You can choose to diplsay them above self-service tickets list (with "Show on helpdesk page")

If you want to control the first bullet point, you need to add a new boolean for that